Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -10559,7 +10559,9 @@ void CPhysicalGeometry::SetWallDistance(CADTElemClass* WallADT, const CConfig* c
 
         WallADT->DetermineNearestElement(nodes->GetCoord(iPoint), dist, markerID, elemID, rankID);
 
-        if (dist < nodes->GetWall_Distance(iPoint)) {
+        if (nodes->GetWall_Distance(iPoint) == 0) dist = 0;
+
+        if (dist < nodes->GetWall_Distance(iPoint) || dist == 0) {
```

Comment:
Here I have forced the assignment of the wall distance even in case the point is on a viscous boundary just to be sure that the other fields are correctly set (markerID etc..). I am not sure if this is necessary or not, I have added it for safety. I have tried removing it and it didn't change, although I have to try with roughness.